src/main/java/hudson/plugins/dimensionsscm/DimensionsAPI.java

```diff
@@ -397,49 +401,61 @@ public final void logout(long key, Run<?, ?> build) {
         return dbCompts;
     }
 
+    @SuppressWarnings("unchecked")
+    List<DimensionsChangeStep> calcRepoDiffsWithChangesets(DimensionsConnection connection, final String projectName,
```

Comment:
Will the `DimensionsAPI` class load successfully and still work if the Java API JARs on the classpath are from Dimensions CM 12 (i.e. if they don't have a definition of the `DimensionsChangeStep` class), or will it throw `NoClassDefFoundError` because of this reference or will it be OK as long as you don't call this method? (The [JVM spec](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-5.html#jvms-5.4) seems to say it would be OK, but worth checking if you haven't).

Reply:
I have tested it with Dimensions 12.2.2 server and new jenkins plugin (with changesets polling support) and all is working fine.